test/Interop/Cxx/stdlib/overlay/std-string-overlay.swift

```diff
@@ -37,22 +37,6 @@ StdStringOverlayTestSuite.test("std::string <=> Swift.String") {
   expectEqual(swift6, "xyz\0abc")
 }
 
-extension std.string.const_iterator: UnsafeCxxInputIterator {
-  // This func should not be required.
-  public static func ==(lhs: std.string.const_iterator,
-                        rhs: std.string.const_iterator) -> Bool {
-#if os(Linux)
-    // In libstdc++, `base()` returns UnsafePointer<Optional<UnsafePointer<CChar>>>.
-    return lhs.__baseUnsafe().pointee == rhs.__baseUnsafe().pointee
-#else
-    // In libc++, `base()` returns UnsafePointer<CChar>.
-    return lhs.__baseUnsafe() == rhs.__baseUnsafe()
-#endif
-  }
-}
-
-extension std.string: CxxSequence {}
```

Comment:
Was removing this line intentional? In `use-std-map.swift` and `use-std-set.swift` they are not removed. I still need to try to reproduce with a pristine upstream checkout, but I have an error in this test because of this:

```
…/swift/test/Interop/Cxx/stdlib/overlay/std-string-overlay.swift:43:12: error: for-in loop requires 'std.string' (aka 'std.__cxx11.__CxxTemplateInstNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE') to conform to 'Sequence'
  for _ in cxx1 {
```

Reply:
Yes, this was intentional: the compiler now automatically conforms `std::string` to `CxxRandomAccessCollection` (which inherits from `CxxSequence`) after this patch, so the manually added extension isn't needed anymore.

The reason why similar extensions for `std::map` and `std::set` were not removed is because we don't currently auto-conform C++ types to `CxxSequence` _unless_ they are random access collections. So for `std::vector` and `std::string` the extension is now redundant, but for `std::set` and `std::map` it is still needed.

Is the test failure happening on macOS?

Reply:
Linux. I haven't double checked macOS. But our Linux C++ library might be the problem in this case. I need to look at this more carefully.